### ISSUE TITLE
Decorated method's type behaves as if unbound

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#2172]
**Describe the bug**
The type of a decorated method behaves as if `self` was not bound to an instance.

**To Reproduce**

```py
#pyright: strict

from typing import Callable

UnboundMethodThatTakesIntAndReturnsStr = Callable[["MyClass", int], str]

def method_decorator(method: UnboundMethodThatTakesIntAndReturnsStr) -> UnboundMethodThatTakesIntAndReturnsStr:
    def wrapper(self: "MyClass", a: int) -> str:
        return "wrapped " + method(self, a)
    return wrapper

class MyClass:
    def __init__(self):
        pass

    @method_decorator
    def do_something(self, a: int) -> str:
        return "foo"

mc = MyClass()

 # do_something's type is as if unbound: 
reveal_type(mc.do_something) # info: Type of "mc.do_something" is "(_p0: Any, _p1: int) -> str"
mc.do_something(123) # error: Argument missing for parameter "_p1"
```

**Expected behavior**
The type of `MyClass().do_something` should be something like `(_p0: int) -> str`. Running the same sample through mypy reveals its type to what is expected: `'def (builtins.int) -> builtins.str'`

**VS Code extension or command-line**
pyright cli and codium extension version 1.1.161

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.162, which I just published. It will also be in the next release of pylance.
